scipy/optimize/_highs/pyHiGHS/src/highs_wrapper.pyx

```diff
@@ -0,0 +1,621 @@
+
+cdef apply_options(dict options, Highs & highs):
+    '''Take options from dictionary and apply to HiGHS object.'''
+
+    # Send logging to dummy file to get rid of output from stdout
+    if options.get('message_level', None) == <int> ML_NONE:
+        highs.setHighsLogfile(NULL)
+        highs.setHighsOutput(NULL)
+    else:
+        # Empty file to send to stdout
+        highs.setHighsLogfile(stdout)
+        highs.setHighsOutput(stdout)
+
+    # Initialize for error checking
+    cdef HighsStatus opt_status = HighsStatusOK
+
+    # Do all the ints
+    for opt in set([
+            b'allowed_simplex_cost_scale_factor',
+            b'allowed_simplex_matrix_scale_factor',
+            b'dual_simplex_cleanup_strategy',
+            b'ipm_iteration_limit',
+            b'keep_n_rows',
+            b'max_threads',
+            b'message_level',
+            b'min_threads',
+            b'simplex_crash_strategy',
+            b'simplex_dual_edge_weight_strategy',
+            b'simplex_dualise_strategy',
+            b'simplex_iteration_limit',
+            b'simplex_permute_strategy',
+            b'simplex_price_strategy',
+            b'simplex_primal_edge_weight_strategy',
+            b'simplex_scale_strategy',
+            b'simplex_strategy',
+            b'simplex_update_limit',
+            b'small_matrix_value',
+    ]):
+        val = options.get(opt.decode(), None)
+        if val is not None:
+            opt_status = highs.setHighsOptionValueInt(opt, val)
+            if opt_status != HighsStatusOK:
+                warn(_opt_warning(opt, val), OptimizeWarning)
+
+    # Do all the doubles
+    for opt in set([
+            b'dual_feasibility_tolerance',
+            b'dual_objective_value_upper_bound',
+            b'dual_simplex_cost_perturbation_multiplier',
+            b'dual_steepest_edge_weight_log_error_threshhold',
+            b'infinite_bound',
+            b'infinite_cost',
+            b'large_matrix_value',
+            b'primal_feasibility_tolerance',
+            b'simplex_initial_condition_tolerance',
+            b'small_matrix_value',
+            b'time_limit'
+    ]):
+        val = options.get(opt.decode(), None)
+        if val is not None:
+            opt_status = highs.setHighsOptionValueDbl(opt, val)
+            if opt_status != HighsStatusOK:
+                warn(_opt_warning(opt, val), OptimizeWarning)
+
+
+    # Do all the strings
+    for opt in set([b'solver']):
+        val = options.get(opt.decode(), None)
+        if val is not None:
+            opt_status = highs.setHighsOptionValueStr(opt, val.encode())
+            if opt_status != HighsStatusOK:
+                warn(_opt_warning(opt, val), OptimizeWarning)
+
+
+    # Do all the bool to strings
+    for opt in set([
+            b'parallel',
+            b'presolve',
+    ]):
+        val = options.get(opt.decode(), None)
+        if val is not None:
+            if val in [True, False]:
+                if val == True:
```

Comment:
Does Cython not support `if val:` with implicit truth check?

Reply:
Yes, but I recall having a very hard time getting it to do what I wanted with boolean checking.  I settled on this rather excruciatingly explicit checking because I was frustrated.  I can go back and see if I can remember what was frustrating me